app/views/play/level/PlayLevelView.coffee

```diff
@@ -197,17 +197,32 @@ module.exports = class PlayLevelView extends RootView
       })
     @setupGod() if @waitingToSetUpGod
 
-    if features.china and e.level.get('slug') is 'magic-rush'
-      @checkForTournamentEnd()
+    if features.china
+      @levelSlug = e.level.get('slug')
+      @setupMandateCheck()
 
-  checkForTournamentEnd: =>
+  setupMandateCheck: =>
     return if @destroyed
     $.get '/db/mandate', (data) =>
       return if @destroyed
-      if data?[0]?.currentTournament isnt 'magic-rush'
-        window.location.href = '/play/ladder/magic-rush'
-      else
-        setTimeout @checkForTournamentEnd, 60 * 1000
+      courseInstanceID = @courseInstanceID or utils.getQueryVariable 'league'
+      if @isTournamentBlocked data?[0], courseInstanceID, @levelSlug
+        unless me.isAdmin()
+          window.location.href = '/play/ladder/'+@levelSlug+(if courseInstanceID then '/course/'+courseInstanceID else "")
+      setTimeout @setupMandateCheck, 60 * 1000
+
+  isTournamentBlocked: (mandate, courseInstanceID, levelSlug) =>
+    return unless mandate
+    tournament = _.find mandate?.currentTournament or [], (t) =>
```

Comment:
Will this just grab the first tournament, always? What if there are multiple tournaments?

Reply:
This looks like it will always return the tournament with the matching courseInstanceID.

Reply:
> Will this just grab the first tournament, always? What if there are multiple tournaments?

One course instance should only have one tournament at the same time. This should be confirmed by administrator.